src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,127 @@
+        isAdmin = true;
+      } catch (Exception exception) {
+        throw new UtilisateurNotFoundException();
+      }
 
-    private final TokenCacheService tokenCacheService;
+    }
 
-    public AuthService(ProprietaireService proprietaireService, JwtHelper jwtHelper, TokenCacheService tokenCacheService) {
-        this.proprietaireService = proprietaireService;
-        this.jwtHelper = jwtHelper;
-        this.tokenCacheService = tokenCacheService;
+    if (proprietaireDTO != null && !proprietaireService.verifyPassword(password, proprietaireDTO.getId())) {
+      throw new ProprietaireNotFoundException();
+    } else if (administrateurDTO != null && !administrateurService.verifyPassword(password, administrateurDTO.getId())) {
+      throw new AdministrateurNotFoundException();
     }
 
-    public AuthLoginResponseDTO loginProprietaire(@NonNull String email, @NonNull String password){
-        ProprietaireDTO proprietaireDTO = proprietaireService.getProprietaireByEmail(email);
+    String uuid = tokenCacheService.getFromCache(id);
+    if (uuid == null) {
+      uuid = UUID.randomUUID().toString();
+      tokenCacheService.addToCache(id, uuid);
+    }
 
-        if(!proprietaireService.verifyPassword(password, proprietaireDTO.getId())){
-            throw new ProprietaireNotFoundException();
-        }
+    String token = jwtHelper.generateToken(email, uuid, isAdmin);
 
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if (uuid == null){
-            uuid = UUID.randomUUID().toString();
-            tokenCacheService.addToCache(proprietaireDTO.getId() ,uuid);
-        }
+    return new AuthLoginResponseDTO(id, token, isAdmin);
+  }
 
-        String token = jwtHelper.generateToken(email, uuid);
+  public AuthLoginResponseDTO auth(@NonNull String token) {
 
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
+    Entry<Long, Boolean> entry = verifyToken(token, false);
 
-    }
+    return new AuthLoginResponseDTO(entry.getKey(), token, entry.getValue());
 
-    public AuthLoginResponseDTO authProprietaire(@NonNull String token){
+  }
 
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-        
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
+  public void logout(@NonNull String token) {
 
-    }
+    Entry<Long, Boolean> entry = verifyToken(token, false);
 
-    public void logoutProprietaire(@NonNull String token){
+    tokenCacheService.addToCache(entry.getKey(), UUID.randomUUID().toString());
 
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
+  }
 
-        tokenCacheService.addToCache(proprietaireDTO.getId(), UUID.randomUUID().toString());
+  public Entry<Long, Boolean> verifyToken(@NonNull String token, boolean shouldBeAdmin) {
+    ProprietaireDTO proprietaireDTO;
+    AdministrateurDTO administrateurDTO;
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'administrateurDTO' must contain no more than '1' consecutive capital letters.